relay-dynamic-config/src/metrics.rs

```diff
@@ -324,7 +325,7 @@ impl MetricExtractionConfig {
     /// The latest version for this config struct.
     ///
     /// This is the maximum version supported by this Relay instance.
-    pub const MAX_SUPPORTED_VERSION: u16 = 3;
+    pub const MAX_SUPPORTED_VERSION: u16 = 4;
```

Comment:
Unfortunately both versions need to be bumped in tandem: If I only bump `TRANSACTION_EXTRACT_MAX_SUPPORTED_VERSION`, old Relays will still set the `metrics_extracted` flag to true when they extract on-demand metrics, which means the crucial legacy transaction metrics (`duration` etc.) will not be extracted.